index.js

```diff
@@ -237,11 +237,19 @@ RedisClient.prototype.create_stream = function () {
             this.stream.destroy();
         }
 
+        // Accept snake_case options and convert them to camelCase
+        var opts = {};
+        var keys = Object.keys(this.connection_options);
+        var key;
+        while (key = keys.pop()) {
+            opts[key.replace(/_([a-z])/g, function (g) { return g[1].toUpperCase(); })] = this.connection_options[key];
+        }
```

Comment:
I don't understand the sacred meaning of conversion to snake_case so I wrote my code right before sending connection options into ```tls``` or ```net```, the same way it was converted. Sorry, this code is not in my style. May be this quest is for the person who has implemented the conversion. Do it as you see fit please.